Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp

```diff
@@ -69,6 +68,7 @@ void CLuaElementDefs::LoadFunctions()
 
         // Element data
         {"getElementData", GetElementData},
+        {"getAllElementData", GetAllElementData},
```

Comment:
Is it possible for you to use `ArgumentParser`? See https://github.com/multitheftauto/mtasa-blue/blob/master/Shared/mods/deathmatch/logic/luadefs/CLuaCryptDefs.cpp for an example.

Let me know if you run into difficulties with the CLuaArguments return type.

Reply:
I'd love to do It with ArgumentParser but It's confusing, couldn't do It sadly

Reply:
I'm using ArgumentParser rn but I don't know the return type, you can help?

Reply:
There does not seem to be an obvious way.

This WON'T work because, while it will put the table on the stack, it will tell Lua that 0 arguments were provided.
```cpp
void GetAllElementData(lua_State* L, CElement* pElement) {
    // logic..
    pVariable->PushAsTable(L);
}
```

Might be nice to create a special return type that automatically wraps returned CLuaArguments, and PushesAsTable when appropriate.

Or support a `LuaReturn<N>` so that we can push our own stuff on the stack and tell the argument parser what we want to do. 

Sorry for wasting your time! I won't block your PR on this change. I'll make a separate requested change that will solve the surface problem.